src/Microsoft.AspNet.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -38,6 +38,34 @@ public Task<int> ReadAsync(ArraySegment<byte> buffer, CancellationToken cancella
             return result;
         }
 
+        public async Task Consume(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Task<int> result;
+            var firstLoop = true;
+            do
+            {
+                result = ReadAsyncImplementation(default(ArraySegment<byte>), cancellationToken);
```

Comment:
Also `Interlocked` is much more expensive than a local var and only one  `100 continue` is sent, so only check it once; though `firstLoop` is probably the wrong name.